File: reasoning/see/complex.jl
# See https://en.wikipedia.org/wiki/Complex_number

expressions = [
    :(sqrt(-1+0im))
    :(exp(π * im) + 1)
    :(sin(im))
]

for expr in expressions
    println("[] :expression \"", replace(repr(expr), "\\" => "\\\\"), "\"; :evaluation \"", eval(expr), "\".")
end
println("")

File: reasoning/see/complex.py
# See https://en.wikipedia.org/wiki/Complex_number

from sympy import sqrt, exp, log, I, pi, cos, sin
from sympy.abc import x

if __name__ == "__main__":
    cases = [
        "sqrt(-1)",
        "sqrt(I)",
        "exp(I*pi) + 1",
        "log(I*x)",
        "cos(I*x)",
        "sin(I*x)"
    ]

    for c in cases:
        print('[] :see "%s = %s".' % (c, eval(c)))
    print("")
